stan/math/prim/eigen_plugins.h

```diff
@@ -53,14 +53,14 @@ struct val_Op{
   //Returns value from a vari*
   template<typename T = Scalar>
   EIGEN_DEVICE_FUNC EIGEN_STRONG_INLINE
-    std::enable_if_t<std::is_pointer<T>::value, reverse_return_t<T>>
+    std::enable_if_t<std::is_pointer<T>::value, const double&>
```

Comment:
Were these changes intentionally part of this PR?

Reply:
Yes, this was a bug. I mention it in the top-level comment: "This also fixes an issue with returning a non-const ref in eigen_plugin.h.".

Basically this should return a const ref, not a non-const.